src/main/java/Duke.java

```diff
@@ -1,10 +1,200 @@
+                        addTask(line,tasks);
+                    }
+                }
+            }
+            Duke.isLoading = false;
+        }
+
+        // ready for User Input
+        else {
+            System.out.println("What can I do for you today?");
+
+            while (true) {
+                Scanner systemScanner = new Scanner(System.in);
+                String systemInput = systemScanner.nextLine();
+
+                if (systemInput.equals("bye")) {
+                    System.out.println("See you later alligator");
+                    break;
+                }
+
+                if (!isQuery(systemInput, tasks)) {
+                    try {
+                        Task addedTask = addTask(systemInput, tasks);
+                        System.out.println("I've added:");
+                        addedTask.describePrint();
+                        System.out.println("You now have " + Task.numberOfTasks + " tasks in the list");
+                    } catch (DukeException e) {
+                        System.out.println(e.getMessage());
+                    }
+                }
+            }
+        }
+    }
+
+    private static boolean isQuery (String input, List<Task> tasks) {
+
+        if (input.contains("done")) {
+            int taskIndex = Integer.parseInt(input.replaceAll("[^0-9]", ""));
+            tasks.get(taskIndex - 1).markAsDone();
+
+            if (!isLoading) {
+                saveCommand(input);
+                System.out.println("This task is done:");
+                tasks.get(taskIndex-1).describePrint();
+            }
+
+            return true;
+        }
+
+        else if (input.equals("list")) {
+            for (int i = 0; i < tasks.size(); i++) {
+                System.out.print(i + 1 + ".");
+                tasks.get(i).describePrint();
+            }
+            return true;
+        }
+
+        else if (input.contains("delete")) {
+            int taskIndex = Integer.parseInt(input.replaceAll("[^0-9]", ""));
+
+            if (!isLoading) {
+                saveCommand(input);
+                System.out.println("I have removed this task:");
+                tasks.get(taskIndex - 1).describePrint();
+                tasks.remove(taskIndex -1);
+                Task.numberOfTasks -= 1;
+                System.out.println("You now have " + Task.numberOfTasks + " tasks in the list");
+            }
+
+            else {
+                tasks.remove(taskIndex -1);
+                Task.numberOfTasks -= 1;
+            }
+
+            return true;
+        }
+
+        return false;
+    }
+
+    private static Task addTask (String input, List<Task> tasks) throws DukeException {
+        String taskType = input.split(" ")[0];
+        String descriptionAndTime = input.replaceAll(taskType, "");
+
+        // check for Task's description
+        if ( (input.contains("todo") || input.contains("deadline") || input.contains("event"))
+                && input.split(" ").length == 1) {
```

Comment:
Avoid complicated expressions. Consider splitting the Booleans up into different steps.